File: Python/Linked List/M-382. Linked List Random Node.py
import random


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def __init__(self, head: Optional[ListNode]):
        self.node_vals = []
        while head:
            self.node_vals.append(head.val)
            head = head.next

    def getRandom(self) -> int:
        return self.node_vals[random.randint(0, len(self.node_vals) - 1)]


# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()

File: Linked List/M-382. Linked List Random Node/M-382. Linked List Random Node.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

class Solution {
  nodeVals: number[];

  constructor(head: ListNode | null) {
    this.nodeVals = [];
    while (head) {
      this.nodeVals.push(head.val);
      head = head.next;
    }
  }

  getRandom(): number {
    const randomNum = Math.floor(Math.random() * this.nodeVals.length);
    return this.nodeVals[randomNum];
  }
}

/**
 * Your Solution object will be instantiated and called as such:
 * var obj = new Solution(head)
 * var param_1 = obj.getRandom()
 */
